charts/gateway-operator/values.yaml

```
@@ -43,3 +43,19 @@ resources:
   requests:
     cpu: 10m
     memory: 128Mi
+
+# Use this section to change the startup parameters of gateway-operator
+featureFlags:
+  # Enable the ControlPlane controller
+  controlPlaneControllerEnabled: true
+  # Enable the Gateway controller
+  gatewayControllerEnabled: true
+  # Enable the DataPlane controller
+  dataPlaneControllerEnabled: true
+  # Enable the DataPlane BlueGreen controller. Mutually exclusive with DataPlane controller
+  dataPlaneBlueGreenControllerEnabled: true
```

Those being mutually exclusive, shouldn't be both enabled by default. I reckon we want to push for BG usage so let's enable that.

```suggestion
  # Enable the DataPlane controller. Mutually exclusive with DataPlane BlueGreen controller
  dataPlaneControllerEnabled: false
  # Enable the DataPlane BlueGreen controller. Mutually exclusive with DataPlane controller
  dataPlaneBlueGreenControllerEnabled: true
```

---

I directly used the configuration values in KGO here. Do we also need to modify the default values in KGO?

---

ref: https://github.com/Kong/gateway-operator/blob/678c485665fece7baaf34c2fe7edfd8f0f6e33fa/modules/cli/cli.go#L37-L38

---

We might update https://github.com/Kong/gateway-operator/blob/ad98c9e851e73c37dcc23439c5fd649baf1b2479/modules/cli/cli.go#L38 as that's not really mutually exclusive but the BG controller takes precedence: whenever it's enabled we disable the standalone DataPlaneReconciler.

---

Got it. Thanks